### ISSUE TITLE
Stuck in redirect loop when token expires

### ISSUE THREAD

**Issue by Achill113** [authts/react-oidc-context#1035]
I am using `react-oidc-context` v2.2.0.

Once a token has expired, despite having the `offline_access` scope and `automaticSilentRenew` set to true, our users end up in a seemingly endless redirect loop where the `addAccessTokenExpired` event is triggered in each loop and continues to generate more and more tokens, but never moves forward to actually be authenticated until localStorage is cleared.

![image](https://github.com/authts/react-oidc-context/assets/10748096/c63c5fc7-a603-40e7-ad4f-77b158c9cccd)

**Comment by pamapa** [comment]
Hard to tell, you somehow made an endless loop. Try enabling logging to find out why. https://authts.github.io/oidc-client-ts/#md:logging

**Comment by Achill113** [comment]
@pamapa thank you for the quick response. I turned on logging, but that didn't seem to give me any more useful information. The only thing in the loop that gets logged is `[UserManager] getUser: user loaded` and then the `addAccessTokenExpired` event is triggered.

![image](https://github.com/authts/react-oidc-context/assets/10748096/dbb82313-c598-47f0-be70-082eeb09bb4e)

**Comment by Achill113** [comment]
@pamapa It seems that `addAccessTokenExpired` is getting triggered; however, `addAccessTokenExpiring` is not. I think this may be because we store our User token in local storage instead of session storage because we had issues with the User token not being available across multiple tabs.

Is there a recommended way to handle this situation?

**Comment by pamapa** [comment]
I do not think it has something to do with local vs session storage. Could it be that the access token you receive is already expired?

**Comment by Achill113** [comment]
I'm not receiving an expired access token, but my refresh tokens are getting rejected by IdentityServer.

**Comment by Achill113** [comment]
I was able to resolve my issue. It came down to accidentally caching the expired token in memory in my frontend, and not grabbing a new one after refreshing.
